### PR TITLE
add reference docs for configuration options

### PR DESCRIPTION
**Pull request by tang-mm**
Add more reference docs from [Goolge drive]( https://docs.google.com/document/d/1-lcFL2XAsgCYpjxwje4fiAdbJwkfDfAZJucBNJ6tRKQ/edit#heading=h.ec3a1r1axbo), including:
- Device connector configuration options
- Supported device connector types
- Testfligner agent configuration options
- Testfligner server configuration options (environment variables)

This PR resolves story: [CERTTF-226](https://warthogs.atlassian.net/browse/CERTTF-226) and [CERTTF-225](https://warthogs.atlassian.net/browse/CERTTF-225)

[CERTTF-226]: https://warthogs.atlassian.net/browse/CERTTF-226?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CERTTF-225]: https://warthogs.atlassian.net/browse/CERTTF-225?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/reference/testflinger-agent-conf.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  output_timeout: 8000
+  execution_basedir: /home/ubuntu/testflinger/rpi-example/run
+  logging_basedir: /home/ubuntu/testflinger/rpi-example/logs
+  results_basedir: /home/ubuntu/testflinger/rpi-example/results
+  logging_level: WARNING
+  job_queues:
+      - rpi4
+      - rpi4-001
+  setup_command: echo Nothing needed for setup
+  provision_command: snappy-device-agent muxpi provision -c /path/to/default.yaml testflinger.json
```
Comment (plars) [inline docs/reference/testflinger-agent-conf.rst:71]: Please replace snappy-device-agent commands everywhere with "testflinger-device-connector" (both will work for now, but we're trying to encourage migration to the new command name)

**Review by plars — CHANGES_REQUESTED** [review]
Looking good, a few small updates below

File: docs/reference/testflinger-agent-conf.rst (hunk 1 above)
Comment (tang-mm) [reply docs/reference/testflinger-agent-conf.rst]: Updated with a new commit `0440672`

**Comment by tang-mm** [comment]
@plars I updated the command and config names per your comments. Could you review it again?
@nancyc12 I added the new `firmware_update` phase (#114 )  into the example. Could you take a look?

File: docs/reference/testflinger-agent-conf.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  execution_basedir: /home/ubuntu/testflinger/rpi-example/run
+  logging_basedir: /home/ubuntu/testflinger/rpi-example/logs
+  results_basedir: /home/ubuntu/testflinger/rpi-example/results
+  logging_level: WARNING
+  job_queues:
+      - rpi4
+      - rpi4-001
+  setup_command: echo Nothing needed for setup
+  provision_command: testflinger-device-connector muxpi provision -c /path/to/default.yaml testflinger.json
+  firmware_update_command: echo Updating device firmware to the latest version
```
Comment (nancyc12) [inline docs/reference/testflinger-agent-conf.rst:74]:
```suggestion
  firmware_update_command: testflinger-device-connector muxpi firmware_update -c /path/to/default.yaml testflinger.json
```

**Review by nancyc12 — CHANGES_REQUESTED** [review]
@tang-mm thanks for making the changes for the firmware_update_command. I added a suggestion for firmware_update_command example.

**Review by nancyc12 — APPROVED** [review]
LGTM +1

**Review by plars — APPROVED** [review]
Sorry, didn't see the update here, +1 from me!

**Comment by plars** [comment]
resolved small merge conflict with the wordlist

**Pull request merged**